File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithDeeplyNestedComplexField/expected/client/EngineProjection.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithDeeplyNestedComplexField.expected.client;

import com.netflix.graphql.dgs.client.codegen.BaseSubProjectionNode;

public class EngineProjection<PARENT extends BaseSubProjectionNode<?, ?>, ROOT extends BaseSubProjectionNode<?, ?>> extends BaseSubProjectionNode<PARENT, ROOT> {
  public EngineProjection(PARENT parent, ROOT root) {
    super(parent, root, java.util.Optional.of("Engine"));
  }

  public EngineProjection<PARENT, ROOT> __typename() {
    getFields().put("__typename", null);
    return this;
  }

  public PerformanceProjection<EngineProjection<PARENT, ROOT>, ROOT> performance() {
     PerformanceProjection<EngineProjection<PARENT, ROOT>, ROOT> projection = new PerformanceProjection<>(this, getRoot());
     getFields().put("performance", projection);
     return projection;
  }

  public EngineProjection<PARENT, ROOT> type() {
    getFields().put("type", null);
    return this;
  }

  public EngineProjection<PARENT, ROOT> bhp() {
    getFields().put("bhp", null);
    return this;
  }

  public EngineProjection<PARENT, ROOT> size() {
    getFields().put("size", null);
    return this;
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/dataClassWithDeeplyNestedComplexField/expected/client/EngineProjection.kt
package com.netflix.graphql.dgs.codegen.cases.dataClassWithDeeplyNestedComplexField.expected.client

import com.netflix.graphql.dgs.client.codegen.InputValueSerializerInterface
import com.netflix.graphql.dgs.codegen.GraphQLProjection
import kotlin.String
import com.netflix.graphql.dgs.codegen.cases.dataClassWithDeeplyNestedComplexField.expected.Generated as ExpectedGenerated
import jakarta.`annotation`.Generated as AnnotationGenerated

@AnnotationGenerated(value = ["com.netflix.graphql.dgs.codegen.CodeGen"])
@ExpectedGenerated
public class EngineProjection(
  inputValueSerializer: InputValueSerializerInterface? = null,
) : GraphQLProjection(inputValueSerializer) {
  public val type: EngineProjection
    get() {
      field("type")
      return this
    }

  public val bhp: EngineProjection
    get() {
      field("bhp")
      return this
    }

  public val size: EngineProjection
    get() {
      field("size")
      return this
    }

  public fun performance(_alias: String? = null,
      _projection: PerformanceProjection.() -> PerformanceProjection): EngineProjection {
    field(_alias, "performance", PerformanceProjection(inputValueSerializer), _projection)
    return this
  }
}
